### ISSUE TITLE
False error "Cannot borrow immutable local variable"

### ISSUE THREAD

**Issue by nlinker** [intellij-rust/intellij-rust#2717]
## Environment

* **Intellij-Rust plugin version:** 
Version: 0.2.0.2101-182

* **Rust toolchain version:** 
stable-x86_64-unknown-linux-gnu
rustc 1.27.2 (58cc626de 2018-07-18)

* **IDE name and version:** CLion 2018.2 Build #CL-182.3684.76, built on July 20, 2018

* **Operating system:** Ubuntu 18.04 LTS

## Problem description

An incorrect error is displayed: "Cannot borrow immutable local variable `boxes[k]` as mutable"
Inspection info:Implements simple cases of borrow checking.

## Steps to reproduce

```
trait Tx {
    fn do_tx(&mut self);
}

struct S<'a>(&'a str);

impl<'a> Tx for S<'a> {
    fn do_tx(&mut self) {
        // technically we want to modify something
        println!("do_tx: {}", self.0)
    }
}

fn do_all(boxes: &mut [Box<dyn Tx>]) {
    let n = boxes.len();
    for k in 0..n {
        boxes[k].do_tx(); // <<<< false error message here
    }
}

fn main() {
    let mut boxes: [Box<dyn Tx>; 3] = [Box::new(S("a")), Box::new(S("b")), Box::new(S("c"))];
    do_all(&mut boxes);
}
```

**Comment by Undin** [comment]
Looks like it was fixed in #2708

**Comment by Undin** [comment]
Duplicate of #2711
